sdk/src/test/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessorTest.java

```diff
@@ -344,44 +390,43 @@ private void unblock() {
     }
   }
 
-  static final class WaitingSpanExporter implements SpanExporter {
-    private final Object monitor = new Object();
+  static class WaitingSpanExporter implements SpanExporter {
 
-    @GuardedBy("monitor")
     private final List<SpanData> spanDataList = new ArrayList<>();
+    private final int numberToWaitFor;
+    private CountDownLatch countDownLatch;
+
+    public WaitingSpanExporter(int numberToWaitFor) {
```

Comment:
Why public?

Reply:
no good reason. It's just test code, but I can tighten it up

Reply:
fixed